internal/flowrate/xrate/monitor.go

```diff
@@ -0,0 +1,98 @@
+package xrate
+
+import (
+	"time"
+
+	"golang.org/x/time/rate"
+
+	legacyimpl "github.com/cometbft/cometbft/internal/flowrate/legacy"
+	cmtsync "github.com/cometbft/cometbft/libs/sync"
+)
+
+type Monitor struct {
+	limiter *rate.Limiter
+
+	// rate.Limiter configuration
+	burst int
+	rate  int64
+
+	// Tokens reserved but not yet used
+	reserved int
+
+	statusMtx cmtsync.Mutex
+	status    legacyimpl.Status
+}
+
+func New(_, _ time.Duration) *Monitor {
+	return &Monitor{
+		limiter: rate.NewLimiter(rate.Inf, 0),
+	}
+}
+
+func (m *Monitor) Limit(want int, maxRate int64, block bool) (n int) {
+	// Update rate limiter parameters, if needed
+	// XXX: this should never change after the first call
+	if maxRate != m.rate {
+		m.limiter.SetLimit(rate.Limit(maxRate))
+		m.rate = maxRate
+	}
+	if want > m.burst {
+		m.limiter.SetBurst(want)
+		m.burst = want
+	}
```

Comment:
This is awkward. I think you're doing it to conform to the existing interface. If that's the case, please add a comment saying this will go away when we get rid of the legacy monitor.